File: model/monetization/SubscriptionModel.py
class SubscriptionModel:
    def __init__(self):
        self.subscriptionId = None  # Long
        self.project = None  # ProjectModel  # This might be an enum
        self.status = None  # SubscriptionStatus  # This might be an enum
        self.renewalType = None  # SubscriptionRenewalType  # This might be an enum
        self.subscriptionPeriod = None  # SubscriptionPeriodOption  # This might be an enum
        self.trial = None  # boolean
        self.plan = None  # PlanModel  # This might be an enum
        self.subscriptionConsumptionRecord = None  # List<SubscriptionConsumptionRecord>
        self.publications = None  # List<RuntimeConfigPublicationModel>
        self.restriction = None  # RestrictionType  # This might be an enum
        self.source = None  # Source  # This might be an enum
        self.startDate = None  # LocalDate  # This might be an enum
        self.endDate = None  # LocalDate  # This might be an enum
        self.nextBillingDate = None  # LocalDate  # This might be an enum
        self.lastBillingDate = None  # LocalDate  # This might be an enum
        self.nextBillingAmount = None  # String
        self.lastBillingAmount = None  # String
        self.lastRenewalDate = None  # LocalDate  # This might be an enum
        self.nextRenewalDate = None  # LocalDate  # This might be an enum
        self.cancellationDate = None  # LocalDate  # This might be an enum
        self.paymentMethod = None  # String
        self.payments = None  # Set<SubscriptionPaymentModel>

File: model/monetization/SubscriptionModel.java
package com.digitinary.dgate.model.monetization;

import com.digitinary.dgate.model.RestrictionType;
import com.digitinary.dgate.model.consumer.ProjectModel;
import com.digitinary.dgate.model.runtimeconfig.RuntimeConfigPublicationModel;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDate;
import java.util.List;
import java.util.Set;

/**
 * Subscription model. a subscription representing the info for a user usage contract details of an api plan.
 *
 * @author Salah Abu Msameh
 * @since 12/05/2024
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@Setter
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class SubscriptionModel {

    private Long subscriptionId;
    private ProjectModel project;
    private SubscriptionStatus status;
    private SubscriptionRenewalType renewalType;
    private SubscriptionPeriodOption subscriptionPeriod;
    private boolean trial;
    private PlanModel plan;
    private List<SubscriptionConsumptionRecord> subscriptionConsumptionRecord;
    private List<RuntimeConfigPublicationModel> publications;
    private RestrictionType restriction;
    private Source source;

    // Subscription tracking fields
    private LocalDate startDate;
    private LocalDate endDate;
    private LocalDate nextBillingDate;
    private LocalDate lastBillingDate;
    private String nextBillingAmount;
    private String lastBillingAmount;
    private LocalDate lastRenewalDate;
    private LocalDate nextRenewalDate;
    private LocalDate cancellationDate;

    // Payment details
    private String paymentMethod;
    private Set<SubscriptionPaymentModel> payments;
}
